test/verify/check-shell-menu

```diff
@@ -75,6 +75,9 @@ class TestMenu(MachineCase):
         b.switch_to_top()
         b.wait_visible("#navbar-oops")
 
+        # clear Oops
```

Comment:
So that makes sense, but it's also kind of a lost opportunity.  Specifically: how do we know for sure that the check for oops is effective?  This seems very much like the sort of thing that might decay at some point in the future if we change the ID of the banner, for example.

It would be cool to have a (separate) test which we expect to fail which shows an oops banner and then exits directly, just to make sure the test works.

The fact that we can clear an oops banner merely by logging out is also somehow suboptimal.  It seems like a lot of tests would logout incidentally at the end, rendering the check ineffective.  It would be cool to be able to detect if there was an oops shown at any point during testing...

Reply:
I'll think about that -- I suppose `@unittest.expectedFailure` should do here, as there is no other way of intercepting a failure in tearDown().

> It would be cool to be able to detect if there was an oops shown at any point during testing...

It would, but it would also add an extra visibility check to *every* individual test step. That seems too expensive to me for such a coarse-grained opportunistic test. Sure, quite a lot of tests use `.logout()` in the middle, but they usually don't at the end.

So what I can do is to add the "no oops" check to `def logout()` as well, and then provide some `self.allow_oops()` helper which sets a flag that tearDown() looks at. WDYT?